### DISCUSSION TITLE
Lesson 1

### DISCUSSION THREAD

**Discussion by 1subatomicParticle** [smartcontractkit/full-blockchain-solidity-course-py#1835] [Q&A] [answered]
Hello, so i have encountered these problems while studying :
**Firstly** there is no Rinkeby testnet anymore in the standard metamask options, so i tried Goerli and I can not find a faucet to get test ETH asap , in most cases I must have an active twitter account or some orther 3rd party account and i still got to wait. So is there a really free faucet for Goerli?
**Secondly**, when i tried to use the Injected Provider Enviroment(used to be called Injected Web3) it doesnt connect to my metamask and i get the following message "No injected provider found. Make sure your provider (e.g. MetaMask) is active and running (when recently activated you may have to reload the page)." so i cannot deploy my first contract.
**Please** take into consideration the following facts before answering, i have a metamask account and add-on connected to Goerli and i have a lot of experience in blockchain transactions and interactions with different wallets in different chains.

**Comment by Niferu** [comment]
Hello @1subatomicParticle 

Rinkeby is depraceted and you should use Goerli now. For faucet use this: https://goerlifaucet.com/. You can also create account on Alchemy because it will be helpful further in course anyway and it will allow you to get more ETH. Regarding second issue few people had same problem and they provided various of solutions. Just go to discussion tab and search for smth like "connect metamask" you will find a lot of questions and solutions.

**Comment by cromewar** [comment, marked as answer]
"Secondly, when i tried to use the Injected Provider Enviroment(used to be called Injected Web3) it doesnt connect to my metamask and i get the following message "No injected provider found. Make sure your provider (e.g. MetaMask) is active and running (when recently activated you may have to reload the page)." so i cannot deploy my first contract."

Try using another browser, an @Neftyr is right Rinkeby was deprecated, please do use Goerli.
